Implement admin account syncing

Turbo87 opened the pull request:
This PR is based on https://github.com/rust-lang/crates.io/pull/7852 and adds a background job that automatically syncs the `users.is_admin` column in the database with the team members of the `crates-io-admins` team on https://github.com/rust-lang/team/ (via https://team-api.infra.rust-lang.org/v1/teams/crates-io-admins.json).

LawnGnome reviewed (approved):
:+1:

I had a pretty good look at the syncing, and it looks good to me.

The one other question I have is: how do we get notified if the sync job fails? I'm conscious of JD's recent issue where it turned out some team sync actions hadn't been happening for a while — how noisy should we make it when the job fails, and similarly, how noisy should the job be when it actually changes something? Team membership changes don't happen very often, so I wonder if we should get notifications _every_ time the job actually updates anything.

Turbo87 commented on Cargo.toml:
```diff
@@ -49,6 +49,7 @@ slow-tests = []
 [dependencies]
 anyhow = "=1.0.79"
 async-trait = "=0.1.77"
+mockall = "=0.12.1"
```
I honestly wonder why I put it there now... this was supposed to be sorted 😅

Turbo87 commented on src/team_repo.rs:
```diff
@@ -0,0 +1,80 @@
+    async fn get_team(&self, name: &str) -> anyhow::Result<Team>;
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct Team {
+    pub name: String,
+    pub kind: String,
+    pub members: Vec<Member>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct Member {
+    pub name: String,
+    pub github: String,
+    pub github_id: i32,
+    pub is_lead: bool,
+}
```
there are similar structs in the team and sync-team repos, but I'm not aware of them being published to a public crate. since it's only seven fields across two structs it's probably sufficient for now to have a copy here?

Turbo87 replied:
it looks like a couple of other projects are depending on https://github.com/rust-lang/team/tree/master/rust_team_data via git, but I'm not particularly keen on adding a git dependency to crates.io

LawnGnome replied:
That's fine; I'm not keen on that either.

What it _might_ mean is that we should have a common crate that everyone can depend upon (including `rust_team_data`), but I don't think we need to do that today.

Turbo87 commented:
As discussed in the team meeting on friday, I've added background job deduplication. If a sync-admin job already exists in the queue then `enqueue-job sync_admin` will not add another one unless `--force` is used.

I've also added notification emails for all of the existing admins when new admins are added or admin access is revoked. Note that admins whose access is revoked will also get these emails, to avoid us unknowingly getting our access revoked by an attacker.

@LawnGnome @walterhpearce could you give this another round of review to see if the concerns are sufficiently addressed? 🙏

bors commented:
:umbrella: The latest upstream changes (presumably 38d7a177936da9ba5e5db7f65d43ecaba7cb9b8f) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

LawnGnome commented on Cargo.toml:
```diff
@@ -129,4 +129,5 @@ crates_io_test_db = { path = "crates_io_test_db" }
 claims = "=0.7.1"
 googletest = "=0.10.0"
 insta = { version = "=1.34.0", features = ["json", "redactions"] }
+regex = "=1.10.2"
```
I'm genuinely a little surprised this is the first time we've used `regex`.

(OK, a quick dig through Git suggests we had it until 2020.)

LawnGnome reviewed (approved):
That looks great. Thanks!
